lib/utils/CrossfilterHistogramPlot.ts

```diff
@@ -101,6 +104,62 @@ export function CrossfilterHistogramPlot(
 				.attr("dx", (_, i) => i === 0 ? "-0.25em" : "0.25em");
 		});
 
+	//~ Background rect for the next section (hover label)
+	svg.insert("rect")
+		.attr("transform", `translate(0,${height - marginBottom})`)
+		.attr("width", 20)
+		.attr("height", 20)
+		.style("fill", "white")
+		.attr("class", "hovered-bg");
+
+	// Value under cursor label
+	const hoveredTick = svg
+		.append("g")
+		.attr("transform", `translate(0,${height - marginBottom})`)
+		.call(
+			d3
+				.axisBottom(x)
+				.tickValues([0])
+				.tickFormat(tickFormatterForBins(type, bins))
+				.tickSize(2.5),
+		)
+		.call((g) => {
+			g.select(".domain").remove();
+			g.attr("class", "gray");
+			g.selectAll(".tick text")
+				.attr("text-anchor", (_, i) => i === 0 ? "start" : "end")
+				.attr("dx", (_, i) => i === 0 ? "-0.25em" : "0.25em")
+				.attr("visibility", "hidden");
+		});
+
+	// `hovered` signal gets updated in mousemove event
+	effect(() => {
+		// const fmt = d3.format("~s");
```

Comment:
can probably remove.

Reply:
Do you mean the commented out d3.format line? Or also the comment about hovered signal?

Reply:
Sorry just the unused code.

Reply:
Also, I had a look into [`d3.format`](https://d3js.org/d3-format#locale_format). I think we probably want something like `d3.format(".2s")` or `d3.format(".3s")` for numbers and whatever `tickFormatterForBins(type, bins)` is for dates.

Honestly, I wonder if just refactoring `tickFormatterForBins` to `formatterForTimeBins(bins)` and then handling the number cases in this file would be more clear.